### PR TITLE
Open API File request for Translator Text API V3

### PR DESCRIPTION
**Pull request by kellyaltom**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by kellyaltom** [comment]
I need to know if there is something wrong I  need to fix.

**Review by marstr — CHANGES_REQUESTED** [review]
Our CI is failing because there is invalid JSON in this PR. Please fix that before we continue reviewing.

https://travis-ci.org/Azure/azure-rest-api-specs/jobs/430223210#L570

**Comment by kellyaltom** [comment]
The error was in two files that I forgot to remove. It should work now.

**Comment by kellyaltom** [comment]
The error was in a file that I forgot to remove. I deleted them.




________________________________

**Comment by kellyaltom** [comment]
should the output file path be the same for all entries in the readme?


________________________________

**Comment by kellyaltom** [comment]
fixed missing ``` before java section

**Comment by marstr** [comment]
@AutorestCI rebuild go

**Review by marstr — CHANGES_REQUESTED** [review]
Instead of adding a new README with AutoRest configuration in the v3.0 folder, you'll need to just modify the [README in the parent directory](https://github.com/Azure/azure-rest-api-specs/blob/acd9ddb21a2b94325d3f5ad823938b40469378a5/specification/cognitiveservices/data-plane/TextAnalytics/readme.md).

**Comment by kellyaltom** [comment]
Readme now in parent directory with autorest config.


________________________________

**Comment by kellyaltom** [comment]
I see the other errors about examples and am working on them.


________________________________

File: specification/cognitiveservices/data-plane/TranslatorText/v3.0/readme.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+> see https://aka.ms/autorest
+
+Configuration for generating Translator Text SDK.
+
+The current release is `release_3_0`.
+
+``` yaml
+
+tag: release_3_0
```
Comment (marstr) [inline specification/cognitiveservices/data-plane/TranslatorText/v3.0/readme.md]: It's okay to have a README in this directory, but it should contain no AutoRest configuration. Please delete this file, or completely replace its contents.

File: specification/cognitiveservices/data-plane/TranslatorText/readme.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+batch:
+  - tag: release_3_0
+```
+
+### Tag: release_1_0 and go
+
+These settings apply only when `--tag=release_1_0 --go` is specified on the command line.
+Please also specify `--go-sdk-folder=<path to the root directory of your azure-sdk-for-go clone>`.
+
+``` yaml $(tag) == 'release_1_0' && $(go)
```
Comment (marstr) [inline specification/cognitiveservices/data-plane/TranslatorText/readme.md]: Copy/paste error? Please update all `release_1_0` to `release_3_0`

File: specification/cognitiveservices/data-plane/TranslatorText/Info.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# Microsoft Translator API V3
+API version 3 only. Translator V2 is depracated.
```
Comment (marstr) [inline specification/cognitiveservices/data-plane/TranslatorText/Info.md]: [nit] "depracated" should be "deprecated" here and in the line below.

**Review by marstr — CHANGES_REQUESTED** [review]

**Comment by kellyaltom** [comment]
added examples in request, added maximum to all array element lengths.

**Review by marstr — DISMISSED** [review]
Looks like we still have some validation errors:
https://travis-ci.org/Azure/azure-rest-api-specs/jobs/432730131#L605

**Comment by kellyaltom** [comment]
Do I need to use the tag 'x-ms-examples' or can I just use 'example' to show examples in the 200 success case.




________________________________

**Comment by kellyaltom** [comment]
Hi Martin, can you help me understand what the errors are? I read the log files but can't make sense of them.

**Review by kellyaltom — COMMENTED** [review]
I have made the fixes

File: specification/cognitiveservices/data-plane/TranslatorText/Info.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# Microsoft Translator API V3
```
Comment (yangyuan) [inline specification/cognitiveservices/data-plane/TranslatorText/Info.md]: Any specific reason we need this file?
This is specifications repo, this kind of announcement should be in documents (or swaggers "info"."description" field).

File: specification/cognitiveservices/data-plane/TranslatorText/readme.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+openapi-type: data-plane
+```
+
+# Releases
+
+### Release 3.0
+These settings apply only when `--tag=release_3_0` is specified on the command line.
+
+``` yaml $(tag) == 'release_3_0'
+input-file: v3.0/TranslatorText.json
```
Comment (yangyuan) [inline specification/cognitiveservices/data-plane/TranslatorText/readme.md]: According to the short document in root page: https://github.com/Azure/azure-rest-api-specs
There should be either a `stable` folder or `preview` folder.
`stable/v3.0/TranslatorText.json`
`preview/v3.0/TranslatorText.json`

File: specification/cognitiveservices/data-plane/TranslatorText/readme.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+This section describes what SDK should be generated by the automatic system.
+This is not used by Autorest itself.
+
+``` yaml $(swagger-to-sdk)
+swagger-to-sdk:
+  - repo: azure-sdk-for-python
+  - repo: azure-sdk-for-java
+  - repo: azure-sdk-for-go
+  - repo: azure-sdk-for-node
```
Comment (yangyuan) [inline specification/cognitiveservices/data-plane/TranslatorText/readme.md]: I didn't see `azure-sdk-for-node` and `azure-sdk-for-ruby` configs...
I think you need to add them or remove these two lines

File: specification/cognitiveservices/data-plane/TranslatorText/readme.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+  add-credentials: true
+  payload-flattening-threshold: 2
+  namespace: azure.cognitiveservices.translatortext
+  package-name: azure-cognitiveservices-translatortext
+  clear-output-folder: true
+```
+``` yaml $(python) && $(python-mode) == 'update'
+python:
+  no-namespace-folders: true
+  output-folder: $(python-sdks-folder)/azure-cognitiveservices-vision-face/azure/cognitiveservices/translatortext
```
Comment (yangyuan) [inline specification/cognitiveservices/data-plane/TranslatorText/readme.md]: azure-cognitiveservices-**vision-face** ?

File: specification/cognitiveservices/data-plane/TranslatorText/readme.md [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+```
+
+### Go multi-api
+
+``` yaml $(go) && $(multiapi)
+batch:
+  - tag: release_3_0
+```
+
+### Tag: release_1_0 and go
```
Comment (yangyuan) [inline specification/cognitiveservices/data-plane/TranslatorText/readme.md]: Should be `release_3_0 and go`?

File: specification/cognitiveservices/data-plane/TranslatorText/v3.0/Info.md [hunk 9]
```diff
@@ -0,0 +1,2 @@
+# Cognitive Services Translator Text SDK
```
Comment (yangyuan) [inline specification/cognitiveservices/data-plane/TranslatorText/v3.0/Info.md]: Please remove this file.

File: specification/cognitiveservices/data-plane/TranslatorText/v3.0/Examples/TranslateSuccess.json [hunk 10]
```diff
@@ -0,0 +1,10 @@
+[
```
Comment (yangyuan) [inline specification/cognitiveservices/data-plane/TranslatorText/v3.0/Examples/TranslateSuccess.json]: The folder name should be in lower case `examples`.

**Review by yangyuan — CHANGES_REQUESTED** [review]

**Comment by yangyuan** [comment]
Hi @kellyaltom,

I saw a few small mistakes. I pointed them out in the review, hope they will be helpful for you to pass the build.

**Comment by kellyaltom** [comment]
Added fixes

**Comment by kellyaltom** [comment]
Fixed examples for 200 success

**Comment by kellyaltom** [comment]
Added Stable directory to the path

**Comment by marstr** [comment]
Howdy @kellyaltom, what's the current status of this PR?

File: specification/cognitiveservices/data-plane/TranslatorText/stable/v3.0/TranslatorText.json [hunk 11]
```diff
@@ -0,0 +1,1105 @@
...
...r intervals beforethe token expires, for instance every 9 minutes.\n\nTo summarize, a client request to the Translator API will include one authorization header taken from the following table:\n\n| Headers       | Description  |\n| ----------    | ----------   |\n| Ocp-Apim-Subscription-key    | Use with Cognitive Services subscription if you are passing your secret key.                               The value is the Azure secret key for your subscription to Translator Text API.                         |\n| Authorization                | Use with Cognitive Services subscription if you are passing an authentication token. The value is the Bearer token: `Bearer <token>`.       |\n\n# Errors\n\nA standard error response is a JSON object with name/value pair named `error`. The value is also a JSON object with properties:\n  * `code`: A server-defined error code.\n  * `message`: A string giving a human-readable representation of the error.\n\nFor example, a customer with a free trial subscription receives the following error once the free quota is exhausted:\n\n```json\n{\n  \"error\": {\n  \"code\":403000,\n  \"message\":\"The subscription has exceeded its free quota.\"\n  }\n}\n```\n"
```
Comment (nschonni) [inline specification/cognitiveservices/data-plane/TranslatorText/stable/v3.0/TranslatorText.json]: beforethe -> before the

File: specification/cognitiveservices/data-plane/TranslatorText/stable/v3.0/TranslatorText.json [hunk 12]
```diff
@@ -0,0 +1,1105 @@
...
...best suited for end-user display.\n      * `numExamples-` An integer representing the number of examples that are available for this translation pair. Actual examples must be retrieved with a separate call to lookup examples. The number is mostly intended to facilitate display in a UX. For example, a user interface may add a hyperlink to the back-translation if the number of examples is greater than zero and show the back-translation as plain text if there are no examples. Note that the actual number of examples returned by a call to lookup examples may be less than `numExamples`, because additional filtering may be applied on the fly to remove \"bad\" examples.\n      * `frequencyCount-` An integer representing the frequency of this translation pair in the data. The main purpose of this field is to provide a user interface with a means to sort back-translations so the most frequent terms are first.\n      \nNote - If the term being looked up does not exist in the dictionary, the response is 200 (OK) but the `translations` list is an empty list.\n  \n# Response Header\nX-RequestId - Value generated by the service to identify the request. It is used for troubleshooting purposes.\n",
```
Comment (nschonni) [inline specification/cognitiveservices/data-plane/TranslatorText/stable/v3.0/TranslatorText.json]: Conjuctions -> Conjunctions

File: specification/cognitiveservices/data-plane/TranslatorText/stable/v3.0/TranslatorText.json [hunk 13]
```diff
@@ -0,0 +1,1105 @@
...
+          "maximum": 100
+        }
+      },
+      "example": {
+        "text": "Bug"
+      }
+    },
+    "DictionaryExampleTextInput": {
+      "type": "object",
+      "description": "Text needed for a dictionary example rquest ",
```
Comment (nschonni) [inline specification/cognitiveservices/data-plane/TranslatorText/stable/v3.0/TranslatorText.json]: rquest -> request

File: specification/cognitiveservices/data-plane/TranslatorText/stable/v3.0/TranslatorText.json [hunk 14]
```diff
@@ -0,0 +1,1105 @@
...
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/Breaksentence": {
```
Comment (nschonni) [inline specification/cognitiveservices/data-plane/TranslatorText/stable/v3.0/TranslatorText.json]: Should this be lowercase `breaksentence` or camel case `BreakSentence` instead?

**Comment by kellyaltom** [comment]
Added all example json files

**Comment by kellyaltom** [comment]
This pull request is passing all tests but is blocked from merging. I have resolved all reviews. Can someone tell me what I need to do to finish this request?

Thanks,
Kelly
________________________________

**Comment by kellyaltom** [comment]
Hi Yang,

I fixed all of the items in your review. Can you review them so the PR can merge.

Thanks,
Kelly
________________________________

**Review by yangyuan — CHANGES_REQUESTED** [review]
Looks good except authentication headers.
Please send the swagger to "Azure API Review Board Core <azureapirbcore@microsoft.com>" for design review and cc me (alias yuanyang)
Please do not expose internal email threads.

FYI, only ADX team can merge, and they will do a full review to make sure this fits azure standards.

File: specification/cognitiveservices/data-plane/TranslatorText/stable/v3.0/TranslatorText.json [hunk 15]
```diff
@@ -0,0 +1,1105 @@
...
+          },
+          {
+            "name": "Authorization",
+            "description": "This is used to pass an authentication token. If you are passing a key and not a token for authentication, then use the other method in the header. **ONE OF THESE METHODS MUST BE USED.**",
+            "in": "header",
+            "type": "string",
+            "required": false
+          },
+          {
+            "name": "Ocp-Apim-Subscription-Key",
```
[the remark this answers is not in the document]
Comment (kellyaltom) [reply specification/cognitiveservices/data-plane/TranslatorText/stable/v3.0/TranslatorText.json]: Added security definitions

[the remark this answers is not in the document]
Comment (yangyuan) [reply specification/cognitiveservices/data-plane/TranslatorText/stable/v3.0/TranslatorText.json]: You also need to remove these headers from operations.
There will effect the behavior of generated SDK.

Have you tested the generated SDK?

**Comment by kellyaltom** [comment]
Ok I will remove, I was not sure if they should stay or not. One method does not require security, Languages.

I did not know the SDKs were available for testing before the pull request was completed. Where are they at? I will test them.

Thanks for your help.

Kelly
________________________________

**Pull request closed without merging**